server/app/services/rpc/node_service_pod_handler.rb

```diff
@@ -1,21 +1,39 @@
 module Rpc
   class NodeServicePodHandler
+    include Logging
+
+    @@lru_cache ||= LruRedux::ThreadSafeCache.new(1000)
```

Comment:
sure, couple megs cache per grid should not be too big "overhead". Doubt many people are actually operating many grids per master anyway, usually only one or couple at max.